src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -0,0 +1,255 @@
+
+    public ItineraryExistence(List<OtpRequest> otpRequests, Itinerary referenceItinerary) {
+        this.otpRequests = otpRequests;
+        this.referenceItinerary = referenceItinerary;
+    }
+
+    /**
+     * Helper function to extract the existence check for a particular day of the week.
+     */
+    public ItineraryExistenceResult getResultForDayOfWeek(DayOfWeek dayOfWeek) {
+        switch (dayOfWeek) {
+            case MONDAY: return monday;
+            case TUESDAY: return tuesday;
+            case WEDNESDAY: return wednesday;
+            case THURSDAY: return thursday;
+            case FRIDAY: return friday;
+            case SATURDAY: return saturday;
+            case SUNDAY: return sunday;
+        }
+        throw new IllegalArgumentException("Invalid day of week provided!");
+    }
+
+    /**
+     * Helper function to set the existence check for a particular day of the week.
+     */
+    private void setResultForDayOfWeek(ItineraryExistenceResult result, DayOfWeek dayOfWeek) {
+        switch (dayOfWeek) {
+            case MONDAY:
+                monday = result;
+                break;
+            case TUESDAY:
+                tuesday = result;
+                break;
+            case WEDNESDAY:
+                wednesday = result;
+                break;
+            case THURSDAY:
+                thursday = result;
+                break;
+            case FRIDAY:
+                friday = result;
+                break;
+            case SATURDAY:
+                saturday = result;
+                break;
+            case SUNDAY:
+                sunday = result;
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Checks whether all checked days of the week are valid.
+     * @return true if all days are either valid (i.e., the day was checked) or null (i.e., the day was not checked).
+     */
+    public boolean allCheckedDaysAreValid() {
+        return (monday == null || monday.isValid()) &&
+            (tuesday == null || tuesday.isValid()) &&
+            (wednesday == null || wednesday.isValid()) &&
+            (thursday == null || thursday.isValid()) &&
+            (friday == null || friday.isValid()) &&
+            (saturday == null || saturday.isValid()) &&
+            (sunday == null || sunday.isValid());
+    }
+
+    /**
+     * @return The first {@link Itinerary} found for the given {@link DayOfWeek}.
+     */
+    public Itinerary getItineraryForDayOfWeek(DayOfWeek dow) {
+        ItineraryExistenceResult resultForDay = getResultForDayOfWeek(dow);
+        return resultForDay != null && resultForDay.isValid() && resultForDay.itineraries.size() > 0
+            ? resultForDay.itineraries.get(0)
+            : null;
+    }
+
+    /**
+     * @return A collection of days of week (and first date found) for which the trip is not possible.
+     */
+    @JsonIgnore
+    @BsonIgnore
+    public Collection<String> getInvalidDaysOfWeek() {
```

Comment:
Since this is no longer returning formatted date strings, it might make more sense just to return a string message instead of a collection of Strings. In other words, the join the strings once again. I originally wrote it to return a collection of dates in case that was needed for some other purpose, but now it seems more specialized for just writing out a message.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/d26baf619770dce9e11667961912b80261de090c..

Reply:
Forgot the javadoc (!)

Reply:
Despair... Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/99a34ede08b01c3684b17a33315869c1e36ce3d4.